src/program.ts

```diff
@@ -3088,6 +3088,13 @@ export abstract class Element {
     return (this.flags & vis) == (other.flags & vis);
   }
 
+  visibilityNoLessThan(other: Element): bool {
+    if (this.is(CommonFlags.Private)) return other.is(CommonFlags.Private);
+    if (this.is(CommonFlags.Protected)) return other.isAny(CommonFlags.Private | CommonFlags.Protected);
+    assert(this.isPublic);
+    return true;
+  }
```

Comment:
I think `public` is a most frequent case, so probably better test it first. wdyt?
```suggestion
  visibilityNoLessThan(other: Element): bool {
    if (this.isPublic) return true;
    if (this.is(CommonFlags.Private)) return other.is(CommonFlags.Private);
    if (this.is(CommonFlags.Protected)) return other.isAny(CommonFlags.Private | CommonFlags.Protected);
    return assert(false);
  }
```
```